src/components/List/List.tsx

```diff
@@ -20,26 +21,63 @@ export default function List(props: React.PropsWithChildren<ListProps>) {
 
   const baseClass = "list";
 
-  let listTag;
+  let listTag,
+    errorText = "";
 
-  React.Children.map(children, function (child: React.ReactElement) {
-    if (child.type !== "li") {
-      throw new Error("Direct children of `List` should be `<li>`s");
-    }
-  });
-
-  if (type === ListTypes.Unordered) {
-    listTag = (
-      <ul id={id} className={bem(baseClass, modifiers, blockName, [className])}>
-        {children}
-      </ul>
-    );
-  } else if (type === ListTypes.Ordered) {
-    listTag = (
-      <ol id={id} className={bem(baseClass, modifiers, blockName, [className])}>
-        {children}
-      </ol>
-    );
+  switch (type) {
+    case "ol":
```

Comment:
The `ListTypes.Ordered` etc enum variables are better for these cases.

Reply:
FIxed.